src/hotspot/cpu/riscv/interp_masm_riscv.cpp

```diff
@@ -700,8 +700,10 @@ void InterpreterMacroAssembler::remove_activation(
     bind(restart);
     // We use c_rarg1 so that if we go slow path it will be the correct
     // register for unlock_object to pass to VM directly
-    ld(c_rarg1, monitor_block_top); // points to current entry, starting
-                                     // with top-most entry
+    ld(c_rarg1, monitor_block_top); // derelativize pointer
+    shadd(c_rarg1, c_rarg1, fp,  c_rarg1, LogBytesPerWord);
```

Comment:
Nit: One redundant space between the 3rd and 4th parameters for each `shadd` call added.

Reply:
Fixed